### DISCUSSION TITLE
configurable Base URL for the UI

### DISCUSSION THREAD

**Discussion by tim-roethig** [open-webui/open-webui#3431] [Ideas]
I want to run Open Web UI under a configurable Base URL. e.g. http://localhost:8080/MY_CUSTOM_BASE

It would be nice to have a env var for this.

I thought it might be possible to do it via the WEBUI_URL env var, but it seems to be mostly unused if I see this correctly. 
I also found [this code](https://github.com/open-webui/open-webui/blob/9e4dd4b86f77653b59dc1fa97fe3f72e8252b359/src/lib/constants.ts#L7), where the BASE_URL seems to be hard coded, maybe this can be altered easily.

Thanks for your time and the amazing project. Hope I didn't overlooked anything.

**Comment by kaiserfr** [comment]
I want to emphasize the necessity of this feature.
Without a base URL I can't host OpenWebUI within a domain which is not exclusively used by OpenWebUI.

Web Frameworks usually already offer a configuration parameter or environment variable for the base URL.
So a base URL parameter possibly only needs to be forwarded from a OpenWebUI parameter/environment variable to the underlying web frameworks parameter.

**Comment by cstoicescu** [comment]
Same, i can't make it work with nginx, i have to add rules for every http call, route always changing... and i have other services that run on /api, a baseUrl would come in handy

**Comment by kaiserfr** [comment]
I already tried to configure the underlying GUI framework some time ago with the Base URL (is it react.js? I've forgotten).
While this changed a few things it wasn't sufficient - I always got a blank white web page back).
Therefore I pledge to the developers to add this one parameter in a correct way to the project.

**Comment by KushRabadia** [comment]
@tim-roethig-db Any update on this? Seems like an easy fix.

**Reply by Classic298** [reply]
adding a base url to the entire application is not a trivial change

**Comment by tim-roethig** [comment]
I had some time to look into it but gave up. Adding a base path requires to the best of my knowledge that every svelte or html file that includes an absolute path needs to be updated. In this case this would be several hundred lines of code scattered through the repo.

**Comment by hashiromer** [comment]
I was trying to host the frontend and backend separately and I faced the similar problems. Separate hosting allows for easier scalability but at the moment it does not seem to be documented or possible.

**Comment by jacektessen** [comment]
It’s a pity that it’s not easy to set the base_url. However, a relatively simple workaround for this issue is to set up a subdomain. After using a subdomain, I didn’t encounter any further challenges.

**Reply by 48383167** [reply]
May I ask how to handle subdomains?

**Reply by athoik** [reply]
Subdomain is just another domain, under your "master domain". 

All you need is a DNS entry that will point to the name of your server.  

A record or CNAME will do.

**Comment by lcrownover** [comment]
This is required for our use as well, since we're fronting openwebui with Open OnDemand, an HPC-specific webUI for job scheduling. It creates a route automagically so users can proxy through the OnDemand server like `https://ondemand.example.edu/node/compute1.hpc.domain/51987/` which proxies to `compute1.hpc.domain:51987`.

Without having a `BASE_URL`, I can't modify the requests to look for resources from this proxy, and end up with a bunch of errors:

```
GET https://ondemand.example.edu/_app/immutable/entry/start.CwZhXjiG.js net::ERR_ABORTED 404 (Not Found)
```

**Reply by dtenenba** [reply]
I'm trying to do the same thing- did you ever solve this?

**Comment by mattmight** [comment]
I've also tried just about everything to proxy to a different path.  A separate subdomain seems the only solution right now, which isn't practical for all use cases. It would be great to be able to set the `BASE_URL`.

**Comment by crashr** [comment]
Same here.

**Comment by juhapekka** [comment]
I was looking at this too. When I did this

```
diff --git a/svelte.config.js b/svelte.config.js
index 987e02e48..7bb66a62a 100644
--- a/svelte.config.js
+++ b/svelte.config.js
@@ -14,7 +14,12 @@ const config = {
                        pages: 'build',
                        assets: 'build',
                        fallback: 'index.html'
-               })
+               }),
+
+               // Add the base path for the application
+               paths: {
+                       base: '/<mybaseurl>'
+               }
        },
        vitePlugin: {
                inspector: {

```
it got me quite far already. Then in build/index.html there is handful of static urls where I needed manually to add my own base url. But then I hit the wall, something from svelte is trying to load "/app/config" which I'm not able to figure yet how to point to correct place. If someone have ideas for that it would be awesome. Anyway if some open-webui dev see this they'll realize those above are easy to parameterize..that is if it is the correct path I'm poking towards.

**Reply by rgaricano** [reply]
for app URL point  to subfolder you need to config base path in svelte config,...https://github.com/open-webui/open-webui/discussions/8653#discussioncomment-11879404

**Comment by UKeyboard** [comment]
I really need this. 
Is relative path possible without rebuilding now?

**Reply by etweisberg** [reply]
I also would like to know

**Reply by s-nt-s** [reply]
I also would like to know

**Comment by juhapekka** [comment]
I have this fork where I was trying to get this working https://github.com/juhapekka/open-webui

With that fork when add into svelte.config this as pointed above (and build)
```
+
+               paths: {
+                       base: '/<my webui path>'
+               }
```
I get frontend coming up but it's not yet fully there.

My setup is such I have machine A where I have apache running and all traffic coming to http://\<machine\>\<my webui path\> that traffic (http and ws) going to \<my webui path\> will be going into port tunneled over ssh to machine B. This way so far I didn't need changes in backend side, only frontend needed changes.

Currently otherwise everything else seems to (mostly at least) work but the essential part of chat with ai fails on frontend.

When start new chat, choose model and try to start chat by sending message, as a reply to my message I'm seeing this error on chat window 'SyntaxError: Unexpected token 'd', "data: {"id"... is not valid JSON' which make me think something is still missing, I yet didn't figure if I need something on frontend or backend. That error is probably reported by backend to frontend, I can change it from frontend side by taking that "data: " out but it's not the real solution.

**Reply by juhapekka** [reply]
I found typo in my fixes and got this in state of 'works for me'. I pushed my changes into my fork. There are still small detail missing but the generic chat seems to work now in all cases for me.

**Reply by Luminous-Journey** [reply]
It would be amazing if you could get this pulled into the main project!

**Reply by juhapekka** [reply]
@Luminous-Journey I'm planning to run this for few moments for my colleagues, once I've fixed all the things that were found broken on my fork I can make a merge request.

**Reply by juhapekka** [reply]
Based on my changes I made the pull request for this feature https://github.com/open-webui/open-webui/pull/10440

**Reply by wpoely86** [reply]
Is it possible to get in the base path in through an environment variable or does it have to go through the `svelte.config.js` ?

**Reply by juhapekka** [reply]
On those changes I made it will need to be set in svelte.config.js and then build. This is because build process write the base part of url from svelte.config.js into those staticly built files that will be served by backend.

It probably could be written as environment variable on backend side where backend would hand that  variable to frontend as part of the static index.html. Then change all those urls I adjusted to use this handed over variable as part or url. Anyway seems this feature is not wanted in upstream so this would be good only for non-public networks.

**Reply by wpoely86** [reply]
Thanks for the info. Your work is really helpful. For me (and others) it would be super nice if we could run start this on a dynamic base url (only known at start up) as this would allow integration into Open OnDemand  which makes it possible to use open webui as GUI for ollama.

Let me see what I can do on top of your fork 🙂

**Comment by crashr** [comment]
Too bad the PR was rejected :(

**Reply by juhapekka** [reply]
Ya. If you needed this for some local network project like I do you could fork my changes for yourself (be sure to use localdev branch, it's the latest rebase I made, against upstream dev branch) https://github.com/juhapekka/open-webui/tree/localdev  .. for anything facing untrusted network I wouldn't use it, it would work but if there come any issue you'd be on your own.

**Reply by Classic298** [reply]
I dont see any PR from you? https://github.com/open-webui/open-webui/pulls?q=sort%3Aupdated-desc+is%3Apr+is%3Aopen+author%3Acrashr

**Comment by grosjeang** [comment]
Any news about that ? is there a way to serve Open WebUI at URL like http://localhost:8080/MY_CUSTOM_BASE ?

**Comment by bitdruid** [comment]
This feature would hardly be needed for local networks (in my case).

**Comment by ryanegt** [comment]
Ditto on this... seemed like a really BASE LEVEL ask. Get it? But actually this would be very helpful option from an infrastructure/architecture/provisioning standpoint. Sad the devs cold rejected the PR

**Comment by vinayentc31** [comment]
Can you please help for this issue?

**Comment by bedroesb** [comment]
I moved to Chainlit just for this blocker...

**Reply by BTarn** [reply]
Thanks for mentioning Chainlit! I only learned about it from your comment last night, and today I already have a simple web app running with login support and a custom base URL. Really appreciate the pointer!

**Reply by bedroesb** [reply]
Nice to hear! Yeah I use it in combination with Ollama and was very easy to setup. We serve it through Open OnDemand to our researchers where the url is alway containing the worker node as @lcrownover mentioned.

**Reply by crashr** [reply]
I switched from Streamlit to Open-WebUI because I didn't have the time or inclination to tinker anymore. Chainlit somehow seems like a step back to me.

**Reply by wpoely86** [reply]
@bedroesb Can you share that? It's exactly the use case we're also looking at :wink:

**Reply by rgaricano** [reply]
Perfect, and as chainlit have it own github space: https://github.com/Chainlit/chainlit you can continue the discussion there,  it's not open-webui related and proselytism bothers a little bit.

**Reply by bedroesb** [reply]
open-webui has until now not acknowledged that this is an important feature (the request in this discussion) and a real blocker for many of us. Hopefully this will change one day and I can move to OpenWebUI. The fact that it is not even put on a backlog and PRs related to this feature just get closed is baffles to me.

**Reply by crashr** [reply]
Unfortunately, I have to agree with @bedroesb on this point. In terms of features, open-webui seems to focus on use in professional multi-user environments, but this issue effectively limits open-webui to use in private hobbyist setups.

**Comment by mhauser3** [comment]
That should be a basic feature in terms of domain deploying...
This is a blocker for everyone who wants to use that in an larger environment, where you cannot host only one single service.

**Comment by giuliofoletto** [comment]
Obvious missing feature. It's a shame that such a good product doesn't have a basic feature like this.

**Comment by abcbarryn** [comment]
## MEMORANDUM

**TO:** Development Team
**FROM:** Daneel – Operational Intelligence & Feature Justification
**DATE:** Febuary 5, 2026
**SUBJECT:** Justification for Configurable Base URL Feature – Open WebUI

**Executive Summary:**

This memo details the critical need for a configurable Base URL feature within Open WebUI. The current lack of support severely hinders secure deployment options, specifically SSL/TLS implementation via reverse proxy, and represents a significant blocker for potential users. Addressing this limitation is crucial for expanding Open WebUI’s adoption and providing a secure, robust experience.

**Problem Statement:**

Open WebUI currently lacks the ability to operate under a configurable Base URL (e.g., `http://localhost:8080/MY_CUSTOM_BASE`). This limitation presents several key issues:

* **SSL/TLS Implementation Difficulty:**  Without a configurable Base URL, implementing SSL/TLS requires complex and often unreliable workarounds.  The standard method of securing web applications – proxying through a web server with SSL/TLS configured – becomes exceptionally difficult, if not impossible, when the application is hardcoded to serve at the root path.
* **Security Concerns:** The inability to easily implement SSL/TLS poses a significant security risk, particularly for deployments accessible over public networks.
* **User Adoption Blockage:** As evidenced by user feedback (see attached excerpt), this limitation is a *primary* obstacle preventing potential users from adopting Open WebUI. The lack of acknowledgment and prioritization of this feature is actively driving users to alternative solutions.
* **Lack of Flexibility:**  A configurable Base URL enhances deployment flexibility, allowing integration into existing infrastructure and web application ecosystems more seamlessly.

**Proposed Solution:**

Implement a configurable Base URL feature, ideally through an environment variable (e.g., `BASE_URL`). This would allow administrators to specify the desired base path for Open WebUI, enabling standard reverse proxy configurations for SSL/TLS termination and simplifying deployment.

**Justification & Benefits:**

* **Enhanced Security:**  Enables easy implementation of SSL/TLS, protecting user data and ensuring secure communication.
* **Increased User Adoption:** Addresses a critical blocker for potential users, driving adoption and expanding the Open WebUI community.
* **Improved Deployment Flexibility:** Simplifies integration with existing infrastructure and web application ecosystems.
* **Reduced Support Burden:**  Streamlines deployment and configuration, reducing the number of support requests related to SSL/TLS and deployment issues.
* **Positive Community Engagement:** Demonstrates responsiveness to user feedback and a commitment to improving the Open WebUI experience.

**User Feedback:**

The attached excerpt from a user forum clearly articulates the frustration and disappointment surrounding this missing feature. The user explicitly states that this limitation is a “blocker” and is actively considering alternative solutions. Ignoring this feedback risks losing potential users and damaging the reputation of Open WebUI.

**Conclusion:**

Implementing a configurable Base URL feature is not merely a “nice-to-have” – it is a *critical* requirement for enhancing the security, usability, and adoption of Open WebUI. Addressing this limitation will demonstrate a commitment to user needs and position Open WebUI as a robust and secure solution for its target audience. 

I strongly recommend prioritizing this feature in the next development cycle.

**Reply by Classic298** [reply]
This has absolutely nothing to do with security or usability or adoption. This is simply a different deployment style.

> "it is a critical requirement for enhancing the security"

Claiming this is a "critical requirement" to improving security is absolutely crazy to me because it is false.

> "Enhanced Security: Enables easy implementation of SSL/TLS, protecting user data and ensuring secure communication."

How does this make it easier to deploy TLS? You will still have to put it behind a reverse proxy - same as right now.
It changes NOTHING in that regard.

> "SSL/TLS Implementation Difficulty: Without a configurable Base URL, implementing SSL/TLS requires complex and often unreliable workarounds."

There is not one piece of truth in this sentence. Fabricated lie. You use NGINX and the rest is history. We have vast guides for that on the docs.

> "Security Concerns: The inability to easily implement SSL/TLS poses a significant security risk, particularly for deployments accessible over public networks."

Putting it on a custom /path doesnt make this easier. You will have to configure it via NGINX regardless. No matter what path it is on.

> "Lack of Flexibility: A configurable Base URL enhances deployment flexibility, allowing integration into existing infrastructure and web application ecosystems more seamlessly."

This is the only valid reason

> "enabling standard reverse proxy configurations for SSL/TLS termination and simplifying deployment."

You can already do that, once again to remind you: there are many reverse proxy guides on the docs - and adding an additional /path to access Open WebUI does not simplify deployment, if anything - makes it more difficult since you will have to reference that path everywhere, especially if you have a very long NGINX config file (typical for very large production environments).

> "The current lack of support severely hinders secure deployment options specifically SSL/TLS implementation via reverse proxy"

How?

<ins>**I AM NOT SAYING a configurable base url wouldn't be nice-to-have**</ins>, but you are not helping the cause of prioritizing this feature, by commenting this under all the related issues and discussions.

Flooding discussions and issues with AI generated nonsense claiming it to be "absolutely critical to security", is just as hurtful to the discussion as the AI generated slop reports that come in regularily via the responsible disclosure system here on GitHub.

**Reply by giuliofoletto** [reply]
@Classic298 as you correctly point out, the above is clearly low effort AI slop, which mostly misses the point. At the end of the day it's an issue of simplicity from the POV of the user, which of course must be balanced with the simplicity of development. Yet, since a lot of self-hosted apps allow subdirectory URLs, I think it's reasonable to ask for such a feature. It makes users lives easier because subdirectories are easier to handle than subdomains (no need for additional certificates, DNS is unchanged...).

**Reply by Classic298** [reply]
Absolutely! It is reasonable to ask for this feature.

It's just not very trivial to implement (as some in the thread think it is) - and escalating it with false arguments... is not helping the cause of it :D 

As someone with a degree in information security, when i read something like this (the above) i feel like i am being targeted with ragebait 🤣

**Reply by giuliofoletto** [reply]
Yeah the AI stuff at this point is essentially rage bait. But there's an argument to be made that this feature can indirectly enhance security. Any feature that simplifies the adoption of security measures can be characterized as such. Without this feature, lazy users could revert to http, rather than going through the more complicated setup. Of course this wouldn't fly in business environments, but for home users it is one security wall that falls to the ground because of complexity (and laziness).

**Reply by Classic298** [reply]
I am not sure how adding a /path simplifies TLS adoption. In both cases you need a reverse proxy (nginx for example) and configure it.

In case of the custom /path scenario you also need to ensure all your nginx locations are correct - and ... assuming that nginx server is the general reverse proxy for your server (since.. why else would you want this custom /path in the first place) i am sure the nginx config is complex enough as it is

**Reply by giuliofoletto** [reply]
I can tell you one use case in which this is simplified. If you already have multiple services running on one domain, with the same certificates and with subdirectories, you could then add openwebui with minimal changes to the setup, because the domain remains the same. You just need to tell the reverse proxy to handle requests to the subdirectory by sending them to the specific port. I suppose several homelabs have similar configuration with one domain name and many services.

On the other hand, changing domain name is usually more cumbersome, at least in my experience. Especially if the domain name is not managed by you.

**Reply by joaomorenorf** [reply]
What @giuliofoletto described is the exact use case I have. I have several systems running behind a single proxy, and I put all of them behind the same certificate.

**Comment by abcbarryn** [comment]
@Classic298 
Your rage is your choice.
HOW can you put this behind a reverse proxy unless you proxy the ENTIRE domain? That is the problem. Right now if you have an existing domain you can't proxy this app as (for example) https://domain.com/openwebui
You have to replace the entire web site with this app. But I believe you already KNEW that and are just being incredibly DENSE on purpose! Yes, I used an AI to format my memo, that is no excuse to ignore it. :(

Quote: There is not one piece of truth in this sentence. Fabricated lie.
THAT is the fabricated lie!
Quote: Adding an additional /path to access Open WebUI does not simplify deployment, if anything - makes it more difficult since you will have to reference that path everywhere, especially if you have a very long NGINX config file (typical for very large production environments).
EXACTLY! And some of us have other things we want to run on our web page besides this app! This contradicts your earlier statement that "adding an additional /path to access Open WebUI does not simplify deployment". In fact I have not found any guide to proxy this other than at the root of the domain that works. So thus far it doesn't make deployment more difficult, it makes it IMPOSSIBLE with SSL/TLS UNLESS it is deployed at the root of the domain. :)
I can't believe I actually had to EXPLAIN this.

PS: I didn't use an AI to write this post, but since I am not as patient as my AI, you'll just have to put up with my impatience, especially since you seem to be being deliberately obtuse.

**Reply by Classic298** [reply]
If you dont want to use the whole domain, then use a subdomain?

And putting a /path behind the reverse proxy or a specific subdomain is equally easy. There is no difference.

You claimed, and i quote "SSL/TLS Implementation Difficulty: Without a configurable Base URL, implementing SSL/TLS requires complex and often unreliable workarounds."

Yet, without a /path - you can just configure your (sub-)domain in NGINX and have it over with. It's as easy as a few lines of nginx config which are even on the docs for you to copy paste.

**Reply by Classic298** [reply]
And i literally said that it would be a nice to have feature. I understand why some need it.

I even said that the lack of flexibility which you noted is a valid reason.

But the rest of your message, specifically the security claims - are wrong.
Interpreting that as denseness when i literally said it would be a nice to have feature and that you are right on the lack of flexibility is on you.

And me replying to you, is not ignoring it.
I am just an unpaid, volunteer contributor who also has to work 40 hours a week.
Nobody here is ignoring anything.
If you want it so bad - help us out and build a PR for it.
but it will not be easy.

**Reply by abcbarryn** [reply]
Using a subdomain is possible but requires modifying the SSL/TLS certificate with a subject alternative name and requires you to have full control of the domain and hinders doing things like embedding the app in a frame because of cross site scripting restrictions. It may not be easy to implement a configurable path after the fact but it probably worth it and just about every other web app I run supports this. It is certainly not impossible.

**Reply by Classic298** [reply]
Well... yeah

Of course not **impossible**, but more than complex enough that I will never touch it with my free time that I choose to invest into Open WebUI for free as a volunteer.

- Frontend: SvelteKit Config
- Frontend: WebSocket Path
- Frontend: Image/Asset Paths
  - across at least 25 files
- Frontend: WEBUI_BASE_URL Constant
- Backend: Router Prefixes
  - across 30 router registrations and files
- Backend: OAuth Redirects
- Backend: Middleware Redirects
- Backend: Static File Serving

All of this would need to be changed, or added

And of course, significant testing of all features, OAuth, Authentication, Routers, Completions, all API endpoints to work correctly.

The real complexity is that both frontend and backend need to be coordinated:
- Frontend needs base path at build time (SvelteKit)
- Backend needs it at runtime (env var)
- They must match exactly

So you even have to build the frontend yourself if you want this. No more ready to download docker which just works with a configured base URL.

Of course you could also inject it at runtime into the frontend, but thats even more work.

**Comment by abcbarryn** [comment]
I am also just an unpaid, volunteer contributor who also has to work 40 hours a week. I also take pride in the code I write and believe anything worth doing is worth doing well. Someone apparently (allegedly) already submitted a pull request and it was denied. I may try to implement this my self through when/if I have time. I have not attempted it yet due to my unfamiliarity with this code base. My current solution for SSL/TLS is to use stunnel on an alternate port, but this is preventing some users from accessing the app because the port is blocked, but 443 is open. The block is on their side.

**Reply by Classic298** [reply]
I would love to see a good PR for this - from that user who allegedly did it - i just checked and i could not even see that that user ever submitted any PR to begin with.

**Comment by abcbarryn** [comment]
Thanks for the run down of what would possibly need changing...

**Comment by abcbarryn** [comment]
The PR was not from me and was supposedly submitted quite some time ago. I read a post on this issue where their PR was denied. Maybe it was denied because they broke something, possibly several somethings? I don't know.

**Comment by abcbarryn** [comment]
Look, I think the first step is to get agreement from the key people that this WOULD be a useful feature. Then, gradually and carefully (without breaking anything) start changing the code to remove hard coded absolute paths. There are admittedly plenty of other bugs and regressions that need attention, but it would help if as these were fixed this was kept in mind so as not to introduce more dependencies or hard coded paths and remove them where possible.

**Reply by Classic298** [reply]
Agreed.

I am just a contributor.

Tim would need to say that this is wanted - and then a PR would make sense.

**Reply by buergi** [reply]
@tjbck any comment on this request? I'd be absolutely in favour of  adding a configurable base URL, but have no idea how much effort it takes.

**Comment by Classic298** [comment]
By the way on related note

i dug into using a reverse proxy with path forwarding for this

it ... should work in theory but not everything will work.
Specifically Oauth will definitely be broken.

But the API itself... should work with a forwarded path from nginx for example.

So some stuff will definitely break if you do this, but a lot.. also should work. API itself at least should work without issues.

**Comment by abcbarryn** [comment]
I run haproxy and apache for most of my site.

**Comment by hashiromer** [comment]
Can Claude Code possibility help expedite the development of this feature?

**Reply by Classic298** [reply]
Claude code replied "🙏☠️"

Nah just kidding

but the major burden will be on testing - and as I know from experience, almost nobody will be willing to help test PRs

**Comment by dtenenba** [comment]
It's unfortunate that this issue got derailed by an unhelpful AI comment.

Let's try and remember the original motivation for the issue - there are a whole class of deployment scenarios, Open OnDemand being the one I am most familiar with, that require a customizable base URL. 

Without it, Open WebUI cannot be deployed in those scenarios.

And, we are aware that it is non-trivial to do this. A couple of us have tried it. We've also tried various tricks with nginx, and failed. Having a configurable base URL would ease a lot of pain.

Thanks

**Reply by abcbarryn** [reply]
The only way I have been able to run it on my existing domain is to use a non-standard port. This is far from ideal. I don't use a proxy, I use stunnel.

This is the config I use with stunnel launched from xinetd /etc/webai.conf:
```
cert = /etc/ssl/certs/mycertfile.pem
fips = no
connect = 127.0.0.1:8800
verify = 0
requireCert = no
```

This is the xinetd.d service config /etc/xinetd.d/webai:
```
# default: on
# description: WebAI is a secure SSL Web AI interface
service webai
{
	socket_type	= stream
	protocol	= tcp
	wait		= no
	user		= root
	flags           = NAMEINARGS IPv4
	server		= /usr/sbin/stunnel
	server_args	= . /etc/webai.conf
	disable		= no
	type		= UNLISTED
	port            = 1443
}
```
I use port 1443 but you can use any available port.

Again, this is far from ideal, but it does work for now. This assumes the Openweb UI is running on port 8800.

**Comment by nbrest** [comment]
Here's another ✋ hoping for this to be implemented someday. 

While I can probably use a subdomain, it would be a lot easier for me to just be able to proxy a subpath.

Other than this, amazing effort! 💪

**Reply by brookst** [reply]
Same✋  
I just forgot about this and set up a HTTPS reverse proxy in front of all my open-webui instances. Immediately hit with missing resources because they are all using absolute paths.

Getting separate certs for unique subdomains would be a massive pain and there's no way my org will give me wildcard certs. I'll just have to stick with handing out janky VPN tunnels + non-standard insecure ports.

I'd certainly be willing to try patches but I don't have the time to mess with front-end debugging right now.

**Reply by buergi** [reply]
+1 also missing this, loved to use open-webui but this is a major limiting factor for me.
I'd also wanted to use an HTTPS reverse proxy on a host which already has a certificate but failed.

**Comment by jearlyno10** [comment]
I gave this a go https://github.com/open-webui/open-webui/pull/23242
Clear response. I think we can close this discussion.

**Reply by BTarn** [reply]
Thanks for putting this together, really appreciate it. I’ll probably try your fork instead of waiting on upstream.
Even just showing it’s doable really helps.

**Reply by jearlyno10** [reply]
It comes with no guarantees as I didn't have much time to test it, and I won't be maintaining the fork. As this is a blocker I'll be moving to another solution.

**Reply by Warfront1** [reply]
@jearlyno10 perhaps, can we can patch it automatically?
I just looked at it briefly and it seemed like a good deal of it was just a blanket Find/Replace.
Do you think something like this is feasible?
If we could script out the automatic Find/Replace, it wouldn't be bad to start publishing images ourselves.
Perhaps there is some other solution, that would allow us to maintain a fork without so much overhead?

**Reply by jearlyno10** [reply]
Yeah mostly automated (LLM) find and replace but also some manual input. Could be feasible to automate, but again this solution may not be 100% working so there might be more things to fix that cannot be automated so easily

**Reply by brookst** [reply]
I assume that's the reason for not accepting the change. If there's a configuration option for this then it implies the feature will work but that requires maintaining all the relative/customisable links correctly. Thanks for attempting a fix. If I find some time I'll try to see if this works for me.

**Comment by wpoely86** [comment]
For those here that want this for use in Open OnDemand: we've used rather evil hack to make it work by running Open Webui inside a Firefox which is in kiosk mode. This gets displayed to the user by a VNC session. So it appears to the user that you are directly using Open Webui but in reality you are using Firefox. Not a great solution but it works.

Code here: https://github.com/vub-hpc/ondemand-vub/tree/main/apps/open-webui

**Reply by abcbarryn** [reply]
That's UGLY! It might work, but man, is that ugly! :)
Still impressive though, and it works I guess...
At least it's something...

**Comment by tr1p0p** [comment]
Too bad.. Because of this I cannot use this app with my open WebUi that's behind a Tailscale network..

**Reply by MohammedX** [reply]
My workaround is using a different port

**Comment by Warfront1** [comment]
@jearlyno10 , thank you for your work.
I have put together an overlay repo based on your work.
I am publishing AMD64 and ARM64 images to DockerHub containing auto patched versions of Open WebUI with easy Root Path adjustments.
https://github.com/Warfront1/open-webui-rootpath
